src/mpid/ch4/netmod/include/netmod.h

```diff
@@ -38,19 +38,23 @@ typedef int (*MPIDI_NM_am_send_hdr_t) (int rank, MPIR_Comm * comm, int handler_i
                                        const void *am_hdr, size_t am_hdr_sz);
 typedef int (*MPIDI_NM_am_isend_t) (int rank, MPIR_Comm * comm, int handler_id, const void *am_hdr,
                                     size_t am_hdr_sz, const void *data, MPI_Count count,
-                                    MPI_Datatype datatype, MPIR_Request * sreq);
+                                    MPI_Datatype datatype, MPIR_Request * sreq, uint8_t protocol);
```

Comment:
I suggest it to be named flags rather than protocol. "protocol" typically refers to handshake sequence, not how data is being transported.

Reply:
Agree on it being named flags. Protocol is one thing we want to express thru this argument. We should leave it open for future extensions.